src/conn/conn_open.c

```diff
@@ -127,7 +127,7 @@ __wt_connection_close(WT_CONNECTION_IMPL *conn)
 	 * conditional because we allocate the log path so that printlog can
 	 * run without running logging or recovery.
 	 */
-	if (FLD_ISSET(conn->log_flags, WT_CONN_LOG_ENABLED) &&
+	if (ret == 0 && FLD_ISSET(conn->log_flags, WT_CONN_LOG_ENABLED) &&
```

Comment:
While this makes intuitive sense, it actually doesn't help the situation at hand.  On develop with this change and running the test, yes we avoid writing that final checkpoint record, but recovery still doesn't run because the true determination is that the close of the metadata handle wrote the `file_sync` records already, and updated the LSN in the turtle file.  So skipping writing the final checkpoint record here doesn't actually matter.  See my `conn_dhandle.c` comment.  Should we keep this change?  It doesn't feel wrong, it just don't solve the problem necessarily either.

Reply:
@sueloverso, as discussed, I'm inclined to keep this one.  It doesn't make sense to write the log record if there has been a failure...